firehose/parsers/flawfinder.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+#
+#   Copyright 2017 David Carlos  <ddavidcarlos1392@gmail.com>
+#   This library is free software; you can redistribute it and/or
+#   modify it under the terms of the GNU Lesser General Public
+#   License as published by the Free Software Foundation; either
+#   version 2.1 of the License, or (at your option) any later version.
+#
+#   This library is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   Lesser General Public License for more details.
+#
+#   You should have received a copy of the GNU Lesser General Public
+#   License along with this library; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301
+#   USA
+
+import sys
+import re
+from subprocess import check_output
+from firehose.model import Message, Point, \
+    File, Location, Generator, Metadata, Analysis, Issue
+
+
+def main():
+    """ Main entry to flawfinder parser """
+    arg_file = sys.argv[1]
+    report = load_file(arg_file)
+    analysis = parse_file(report)
+    sys.stdout.write(str(analysis.to_xml()))
+    sys.stdout.write('\n')
+
+def load_file(arg_file):
+    """ Loads the Flawfinder report in memory
+
+    :arg1: file passed by cli
+    :returns: file loaded in memory
+    """
+    try:
+        return open(arg_file, 'r')
+    except Exception as e:
+        raise e
+
+
+def parse_file(loaded_file):
+    """ Parser loaded flawfinder output
+
+    :loaded_file: flawfinder report, loaded in memory
+    :returns: Firehose Analysis object, representing the final XML.
+    """
+
+
+    line = loaded_file.readline()
+    flawfinder_version = flawfinder(line)
+    generator = Generator(name='flawfinder',
+                          version=flawfinder_version)
+    metadata = Metadata(generator, None, None, None)
+    analysis = Analysis(metadata, [])
+
+    flawfinder_data = []
+    pattern = "\([a-z]*\)"
+    pattern2 = "\[[0-9]\]"
+    prog = re.compile(pattern)
+    prog2 = re.compile(pattern2)
+
+
+    while line:
+        if line != "\n" and (prog.search(line) and prog2.search(line)):
+            data = {}
+
+            data["cwe"] = ''
+            data["path"] = line.split(":")[0]
+            data["line"] = line.split(":")[1]
+
+            message_line = loaded_file.readline()
+            message = ""
+            while message_line != "\n" and not \
+            (prog.search(message_line) and prog2.search(message_line)):
+                message += " " + message_line
+                extracted_cwe = get_cwe(message_line)
+                if extracted_cwe != '':
+                    data["cwe"] = extracted_cwe
+                message_line = loaded_file.readline()
+            data["message"] = message;
+            line = message_line
+            flawfinder_data.append(data)
+        else:
```

Comment:
I'm finding it hard to follow this two-stage parsing.  If I'm reading things right, the code first reads the lines, building up a list of dicts (``flawfinder_data``).  There's then a second loop over that list, creating ``Issue`` objects and adding them to ``analysis.results``.

Wouldn't it be much simpler to have one loop, and append ``Issue`` objects directly to ``analysis.results``, without the ``data`` dict i.e. work directly with an ``Issue`` object instead.

Hope that makes sense; I may have misread this code.